ocaml/otherlibs/systhreads/st_stubs.c

```diff
@@ -837,31 +851,26 @@ CAMLprim value caml_thread_yield(value unit)        /* ML */
   struct caml_locking_scheme* s;
 
   s = atomic_load(&caml_locking_scheme);
-  if (s->count_waiters != NULL && s->yield != NULL) {
-    if (s->count_waiters(s->context) == 0) return Val_unit;
-
-    /* Do all the parts of a blocking section enter/leave except lock
-       manipulation, which we'll do more efficiently in st_thread_yield. (Since
-       our blocking section doesn't contain anything interesting, don't bother
-       with saving errno.)
-    */
-    caml_raise_async_if_exception(caml_process_pending_signals_exn(),
-                                  "signal handler");
-    caml_thread_save_runtime_state();
-    s->yield(s->context);
-    if (atomic_load(&caml_locking_scheme) != s) {
-      /* The lock we have is no longer the runtime lock */
-      s->unlock(s->context);
-      acquire_runtime_lock();
-    }
-    caml_thread_restore_runtime_state();
-    caml_raise_async_if_exception(caml_process_pending_signals_exn(),
-                                  "signal handler");
-  } else {
-    caml_enter_blocking_section();
-    st_fallback_yield();
-    caml_leave_blocking_section();
+  if (s->can_skip_yield != NULL && s->can_skip_yield(s->context))
```

Comment:
Since `can_skip_yield` can just be the constant `false` function, do we really need this `Null` check?

Reply:
I think NULL is more idiomatic in a language without closures. Asking API users to implement the constant false function is a bit unkind in C.